apps/fz_http/lib/fz_http/devices.ex

```diff
@@ -23,6 +23,10 @@ defmodule FzHttp.Devices do
 
   def get_device!(id), do: Repo.get!(Device, id)
 
+  def get_device({:public_key, public_key}) do
```

Comment:
It may be a tad more clear to use `get_by_public_key(public_key)` as I use this pattern in `users.ex` for `get_by_email`.

Reply:
That's fine too.

Reply:
Ended up removing the function, since when `@events_module.delete_device` is called the `Device` is already removed from the DB so we need to get the `device` instead of only the pubkey. So I simply pass the whole device now and removed the `FzHttp.Events.delete_device` for pub_key.